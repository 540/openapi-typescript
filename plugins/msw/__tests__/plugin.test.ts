import { generateCode, loadFixture } from "./utils.js";

describe("msw plugin", () => {
  const testCases = [
    {
      fixture: "minimal-openapi-no-paths.json",
      description: "should generate correct TypeScript interfaces for empty OpenAPI spec",
    },
    {
      fixture: "simple-path-201-no-body.json",
      description: "should generate correct TypeScript interfaces for basic API endpoints",
    },
    {
      fixture: "path-with-request-body.json",
      description: "should generate correct TypeScript interfaces for endpoints with request bodies",
    },
    {
      fixture: "path-with-200-response-object.json",
      description: "should generate correct TypeScript interfaces for endpoints with complex responses",
    },
  ];

  it.each(testCases)("$description", async ({ fixture }) => {
    const spec = loadFixture(fixture);

    const generatedCode = await generateCode(spec);

    expect(generatedCode).toMatchSnapshot();
  });
});
