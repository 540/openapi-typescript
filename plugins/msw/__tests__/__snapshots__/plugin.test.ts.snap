// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`msw plugin should generate correct TypeScript interfaces for basic API endpoints 1`] = `
"import { createOpenApiHttp as baseCreateOpenApiHttp } from "openapi-msw";
export const createOpenApiHttp: typeof baseCreateOpenApiHttp<paths> = baseCreateOpenApiHttp;
export interface paths {
    "/resource": {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        get?: never;
        put?: never;
        /** Create a new resource */
        post: operations["createResource"];
        delete?: never;
        options?: never;
        head?: never;
        patch?: never;
        trace?: never;
    };
}
"
`;

exports[`msw plugin should generate correct TypeScript interfaces for empty OpenAPI spec 1`] = `
"import { createOpenApiHttp as baseCreateOpenApiHttp } from "openapi-msw";
export const createOpenApiHttp: typeof baseCreateOpenApiHttp<paths> = baseCreateOpenApiHttp;
export type paths = Record<string, never>;
"
`;

exports[`msw plugin should generate correct TypeScript interfaces for endpoints with complex responses 1`] = `
"import { createOpenApiHttp as baseCreateOpenApiHttp } from "openapi-msw";
export const createOpenApiHttp: typeof baseCreateOpenApiHttp<paths> = baseCreateOpenApiHttp;
export interface paths {
    "/resource": {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        /** Get a resource */
        get: operations["getResource"];
        put?: never;
        post?: never;
        delete?: never;
        options?: never;
        head?: never;
        patch?: never;
        trace?: never;
    };
}
"
`;

exports[`msw plugin should generate correct TypeScript interfaces for endpoints with request bodies 1`] = `
"import { createOpenApiHttp as baseCreateOpenApiHttp } from "openapi-msw";
export const createOpenApiHttp: typeof baseCreateOpenApiHttp<paths> = baseCreateOpenApiHttp;
export interface paths {
    "/resource": {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        get?: never;
        put?: never;
        /** Create a resource with request body */
        post: operations["createResourceWithBody"];
        delete?: never;
        options?: never;
        head?: never;
        patch?: never;
        trace?: never;
    };
}
"
`;
